src/Libraries/CoreNodeModels/Formula.cs

```diff
@@ -20,262 +20,40 @@ namespace CoreNodeModels
-                    {
-                        Arguments = inputs.Select(AstFactory.BuildParamNode).ToList()
-                    },
-                ReturnType = TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.Var),
-                FunctionBody =
-                    new CodeBlockNode
-                    {
-                        Body =
-                            new List<AssociativeNode>
-                                {
-                                    AstFactory.BuildReturnStatement(
-                                        AstFactory.BuildFunctionCall(
-                                            backingMethod,
-                                            new List<AssociativeNode>
-                                            {
-                                                AstFactory.BuildStringNode(FormulaString),
-                                                AstFactory.BuildExprList(
-                                                    InPorts.Select(
-                                                        x =>
-                                                            AstFactory.BuildStringNode(x.Name) as
-                                                            AssociativeNode).ToList()),
-                                                AstFactory.BuildExprList(
-                                                    inputs.Select(AstFactory.BuildIdentifier)
-                                                        .Cast<AssociativeNode>()
-                                                        .ToList())
-                                            }))
-                                }
-                    }
-            };
-
-            if (IsPartiallyApplied)
-            {
-                return new AssociativeNode[]
-                {
-                    functionDef,
-                    AstFactory.BuildAssignment(
-                        GetAstIdentifierForOutputIndex(0),
-                        AstFactory.BuildFunctionObject(
-                            functionDef.Name,
-                            InPorts.Count,
-                            Enumerable.Range(0, InPorts.Count).Where(index=>InPorts[index].IsConnected),
-                            inputAstNodes))
-                };
-            }
-            else
+            if (!conversionFailed)
             {
-                UseLevelAndReplicationGuide(inputAstNodes);
-
-                return new AssociativeNode[]
-                {
-                    functionDef,
-                    AstFactory.BuildAssignment(
-                        GetAstIdentifierForOutputIndex(0),
-                        AstFactory.BuildFunctionCall(
-                            functionDef.Name,
-                            inputAstNodes))
-                };
+                newNode.Attributes["CodeText"].Value = convertedCode;
+                //(node as CodeBlockNodeModel).FormulaMigrationWarning(Resources.FormulaMigrated);
```

Comment:
is this not accessible from here?

Reply:
No, over here I don't have a handle to the Nodemodel, just the xml node. Trying to figure this one out as the last thing to do here.